plugins/resources/replication/librepl.cpp

```diff
@@ -215,22 +215,60 @@ extern "C" {
         return result;
     }
 
+    bool is_self_in_hier(
+        irods::resource_plugin_context& _ctx,
+        const irods::hierarchy_parser &_parser) {
+        std::string name;
+        irods::error ret = _ctx.prop_map().get<std::string>(irods::RESOURCE_NAME, name);
+        if (!ret.ok()) {
+            THROW(ret.code(), ret.result());
+        }
+        return _parser.resc_in_hier(name);
+    }
+
     irods::error get_selected_hierarchy(
         irods::resource_plugin_context& _ctx,
         std::string& _hier_string,
         std::string& _root_resc ) {
         irods::hierarchy_parser selected_parser;
         std::string operation;
 
-        irods::error ret = _ctx.prop_map().get<irods::hierarchy_parser>( HIERARCHY_PROP, selected_parser );
-        if ( !ret.ok() ) {
-            return PASSMSG(
-                       (boost::format(
-                        "[%s] - Failed to get the parser for the selected resource hierarchy.") %
-                        __FUNCTION__).str(), ret );
+        try {
+            bool resc_hier_in_kw = false;
+            irods::file_object_ptr file_obj = boost::dynamic_pointer_cast<irods::file_object >( _ctx.fco() );
+            if (file_obj->l1_desc_idx() > 0) {
+                char* hier_str = getValByKey(&L1desc[file_obj->l1_desc_idx()].dataObjInp->condInput, RESC_HIER_STR_KW);
+                if (!hier_str) {
+                    return ERROR(SYS_INTERNAL_NULL_INPUT_ERR,
+                                 (boost::format(
+                                  "[%s] - No hierarchy string found in keywords for file object.") %
+                                  __FUNCTION__).str().c_str());
+                }
+                selected_parser.set_string(std::string(hier_str));
```

Comment:
Is wrapping `hier_str` in `std::string` required?

Reply:
Doesn't look like it. Removing!